src/lxc/terminal.c

```diff
@@ -584,6 +584,18 @@ static int lxc_terminal_peer_proxy_alloc(struct lxc_terminal *terminal,
 		goto on_error;
 	}
 
+	ret = fcntl(terminal->proxy.master, F_SETFD, FD_CLOEXEC);
```

Comment:
I always wonder - given the fds come from `openpty` where you don't pass flags and don't exactly know which flags they have, if this shouldn't be a read-modify-write as stated in the `fcntl(ep)` manpage's `Application Usage` section.

Reply:
You mean retrieve the flags, set the `FD_CLOEXEC` flag and pass the flags back down?

Reply:
Care to push a patch on top of my branch?

Reply:
Hm, or send me one :D